snapshots/devmapper/README.md

```diff
@@ -0,0 +1,46 @@
+## Devmapper snapshotter
+
+Devmapper is a `containerd` snapshotter plugin that stores snapshots in ext4-formatted filesystem images
+in a devicemapper thin pool.
+
+## Setup
+
+To make it work you need to prepare `thin-pool` in advance and update containerd's configuration file.
+This file is typically located at `/etc/containerd/config.toml`.
+
+Here's minimal sample entry that can be made in the configuration file:
+
+```
+[plugins]
+  ...
+  [plugins.devmapper]
+    pool_name = "containerd-pool"
+    base_image_size = "128MB"
+  ...
+```
+
+The following configuration flags are supported:
+* `root_path` - a directory where the metadata will be available (if empty
+  default location for `containerd` plugins will be used)
+* `pool_name` - a name to use for the devicemapper thin pool. Pool name
+  should be the same as in `/dev/mapper/` directory
+* `data_device` - path to the data volume that should be used by the thin pool
+* `meta_device` - path to the metadata volume that should be used by the thin-pool
+* `data_block_size` - the size of allocation chunks in data file, between 128
+  sectors (64KB) and and 2097152 sectors (1GB) and a multiple of 128 sectors (64KB)
+* `base_image_size` - defines how much space to allocate when creating the base device
+
```

Comment:
Hi @mxpv,

IMHO, we'd better just keep snapshotter taking care of one thing - taking and managing snapshots,
and leave the management of thin-pool (create or assemble) to a storage-manager tool independent of containerd.

Thus, snappshotter only takes in `pool_name`, check on it and fails to continue if the pool is not ready, leaving
the storage tool to handle storage magement details such as using raid or dmcache under the thin-pool, monitor the fullness and auto-extend the thin-pool, etc.

Reply:
Fixed